app/public/js/common/SCapiService.js

```diff
@@ -186,6 +186,86 @@ app.service('SCapiService', function($http, $window, $q, $log, $state, $statePar
     };
 
     /**
+     * Responsible to get track details
+     * @return {[object]} data response
+     */
+    this.getTrackDetails = function(songId) {
+        var url = 'https://api.soundcloud.com/tracks/' + songId + '.json?&oauth_token=' + $window.scAccessToken,
+            that = this;
+        return $http.get(url)
+            .then(function(response) {
+                if (typeof response.data === 'object') {
+                    return response.data;
+                } else {
+                    // invalid response
+                    return $q.reject(response.data);
+                }
+
+            }, function(response) {
+                // something went wrong
+                return $q.reject(response.data);
+            });
+    };
+
+
+    /**
+     * Responsible to get comments for track
+     * @return {[object]} data response
+     */
+    this.getTrackComments = function(songId) {
+        var url = 'https://api.soundcloud.com/tracks/' + songId + '/comments.json?linked_partitioning=1&limit=30&oauth_token=' + $window.scAccessToken,
+            that = this;
+        return $http.get(url)
+            .then(function(response) {
+                if (typeof response.data === 'object') {
+                    if ( response.data.next_href !== null || response.data.next_href !== undefined ) {
+                        that.next_page = response.data.next_href;
+                    }
+                    return response.data;
+                } else {
+                    // invalid response
+                    return $q.reject(response.data);
+                }
+
+            }, function(response) {
+                // something went wrong
+                return $q.reject(response.data);
+            });
+    };
+
+    /**
+     * Responsible to get user favorite tracks ids
+     * @return {[object]} data response
+     */
+    this.getFavoritesIds = function(next_href) {
```

Comment:
The API response comes with `user_favorite` so let's just use that.. if it's null or empty nothing will happen, no need to all that extra work/code.